### ISSUE TITLE
Convention Framework Rework/Proposal: Universal Chop Move Principles

### ISSUE THREAD

**Issue by pianoblook** [hanabi/hanabi.github.io#479]
I wrote a new framework [doc](https://docs.google.com/document/d/1Y9SSdwKZjjvkG4E80T8SNoRawkYdf14zNw204EuURiU/edit) which lays out a few proposed changes to how certain Chop Move situations could be handled differently. kimbifille, piper, and I have been developing & refining this system since around summer, and we're super excited to share it in its final form.

If you want to understand the case for these changes in detail, please read the doc. It's fairly long but hopefully interesting to anyone whose interested in considering the merits of adopting this system vs not. For the sake of this github issue, though, I'll at least copy over the intro and the three main proposed _Principles_.

> This document proposes three new Universal Chop Move Principles that would result in a fairly significant simplification of how a variety of clues would be interpreted. Both for scorehunting and in competition, we’ve found these changes to (a) be simple and intuitive, (b) provide better tempo and efficiency across a wide variety of situations, and (c) provide a groundwork for several exciting new opportunities.

**1. Unnecessary Chop Move Principle**
If a single, playable card is ever _Chop Moved_, the next player should immediately blind-play a card. Afterwards, the clued player can now always consider this card as being _Pulled._

**2. Bad Chop Move Ejection Principle**
If any _Chop Move_ action would result in nothing but Trash being _Chop Moved_, the next player is always being called to immediately _Eject_.

**3. 5 Chop Move Principle**
Until the _End-Game_, except for valid stalling situations, all one-off-chop 5 rank clues are to be promised as _5 Chop Moves_, regardless of the identity of the 5.

For what it's worth, adopting these would technically lead to a *decrease* in overall conventions: 5 Double Pull, Trash Pull, BCME w/ 5s, BCME w/ 1s would all just become natural extensions. Plus, all the additional Ejection opportunities that arise would also not need to have their own convention listed.

That said, note that the last section of the doc also lays out 5 new extensions/convention proposals that build off of these core principles. I separated them because none of them would be necessary to approve in order for the basic system to work smoothly. Let me know if you'd rather me open a separate Issue for them independently, btw

### In Game Examples ###

5CM Double: 
https://hanab.live/replay/403684#17
https://hanab.live/replay/329695#18
https://hanab.live/replay/334096#22
https://hanab.live/replay/295971#37

Trash Double Pull:
https://hanab.live/replay/380594#45 
https://hanab.live/replay/352324#42
https://hanab.live/replay/310831#36
https://hanab.live/replay/290377#37
https://hanab.live/replay/274009#39 (half credit since all critical are seen)

Trash Delayed Double Pull:
https://hanab.live/replay/413161#40

Midgame Multi-TCM Double Pull:
https://hanab.live/replay/381468#24
https://hanab.live/replay/295934#37

TCCM Double Pull:
https://hanab.live/game/17625#33
https://hanab.live/replay/316999#32

(attempted stale TCM Finesse
https://hanab.live/replay/396259#15

1-away 5 BCME
https://hanab.live/replay/310789#39

BTCCME:
https://hanab.live/replay/305868#34

RBOCME:
https://hanab.live/replay/289454#33

**Comment by Jayhui-q** [comment]
This is super neat! I am very for principles that help unify pieces in the conventions and provide a bit more intuition to it.

This line here is quite inspiring for this:
> after seeing overwhelmingly positive results 

Do you think you could share a few replays?

It was awesome and super fun to read all of these thoughtful points. Overall, I think there's definitely a lean towards faster play and away from flexible play. I wouldn't be surprised if this inspires some new convention changes. I've put a bunch of thoughts I wrote while reading below:
______

> When a 5 is clued one off-chop after Early Game, the next player needs to go through a multi-step process. Is it a Chop Move or a Finesse?

I think 5cm's are actually pretty well defined. The current documentation for [5's Chop move](https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#the-5s-chop-move-5cm) and [BCME](https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#bad-chop-move-ejection-with-5s) make it pretty clear that unless the context or a stall situation says otherwise, a 5 to a 1-off-chop is a CM to the clue recipient always. 

The next player who can react will ask: Is the cm'ed card trash? If so it's a BCME if the 5 requires two blind-plays, and it's a finesse if it's one-away. 

Seems pretty straight-forward? Maybe I am missing something about this, but I have never struggled with a 5 clue.
________

> In practice, of course, this just means such a clue [a trash chop move that chop moves good and bad cards] could never be given with the conventions as is.

Don't we all do this all the time? We just deal with it and clarify which one was the good one later. I may or may not be alone in this practice.
_______

> Unfortunately, Hyphen-ated conventions do not permit Bob to react at all in either case [of tempo clue chop moving a playable or a trash card]. 

In my experience, Bad TCCM ejections have caused more strikes than they were worth. I don’t have enough time with it though. This is 100% just from a few, limited games. In complicated variants, sometimes I gave a tempo clue that would be a tccm, even though it cm'ed a random card, just for tempo purposes. If you have had a lot of Bad TCCM ejection success, do share!
_____

> If a single, playable card is ever Chop Moved, the next player should immediately blind-play a card. Afterward, the clued player can now always consider this card as being Pulled.

The strength of the always and ever is super weird to me. First, I think it has to be "directly playable" rather than a delayed play clue, as pulls are. The 5 might get chopped, otherwise. I don't think we would rather give the direct play and then give the 5, because it's less efficient, and we would have to place a litany of restrictions, such as "if there are more than 1 clues, and Alice clues Cathy, and Bob is not currently finessed."

Even if it is directly playable, I do still have concerns. 

In a 4p Rainbow Variant,
- b2 is on the stacks
- r4, g3 is in the trash
- Bob has an unknown 4 (it is b4)
- Cathy's hand is b4 - x - r5 - b3

I think Alice giving a 5cm is far superior to blues as a direct play clue. In this framework, 5s would cause Bob to blind-play. 

Alternatively, Bob has a clued m1 with rank (and no color) in hand. The blue clue to chop is now ambiguously m2. the 5 falls. 

In addition, more stipulations for ambiguity must be made. If Donald has m1 that is finessable, the blue clue is also bad. 

There is a point in the document where you mention:

> The rare exception to this is when a 5 Chop Move might have been a necessary way

I do think that the example you provided for the context read is a bit too simple, especially if there are unknowns in a more complicated variant.

_________

> currently-unactionable ones such as OCM,

This one is funny. I have so wanted to do a bad order chop move ejection for a while now. The likelihood of having 3 trash or 1s and then the ejection slot of Bob being playable and not a 1, is remarkably low. I haven't done it yet!
_______

> If any Chop Move action would result in nothing but Trash being Chop Moved, the next player is always being called to immediately Eject.

Agreed, mostly..... I think if the 5 is 1-away-from-playable, isn't the simplest read that it's a finesse/bluff? (as 5BCME is currently written?)  I only say this because the finesse position is more common, in my view. 
_______

> [Regarding 5s finesses/bluffs] Most of the time, the same Bluff or Finesse can still be triggered by cluing color to the 5.

Agreed, mostly! In No Variant, yes. In any multi-color, I'm not so certain.

Something to think about: if a two-blind-plays-required 5 is focused with color, in this case, you eject with 5CE. If the same is focused with rank and it cms trash, you also eject. Just an interesting thought that the same card is used for an ejection in two different ways. It may also explain why I rarely do a 5BCME. Maybe, this is a reason why we should move something to not be an ejection? Like how 5ND was originally made over 5NE because you could otherwise use color. Maybe it really should be 5BCMD (!!!!! if this is the outcome of this whole conversation that would be wild!). Anyway, just some ruminating thoughts.
______

> because frankly the ambiguity of Unnecessary Trash CMs (UTCM)

Is it ambiguous? Avanderwalde, Rkass, and I do this fairly often, usually to much success! 
Would be very curious if others have not used the UTCM much.
______

> BCME (w/ 5s) become much more frequently available.

Hesitating on the "much more" part of this benefit. It's only when the 5 is playable or one-away, AND we were willing to let a playable card get to the ejection slot. Having 5s be able to be used for a bluff/finesse actually short-circuits the need to eject. 
______

> Mid-Game Trash Double Pulls are now incredibly easy to perform.

Nice. Agreed, this is a neat benefit. I do want to weigh against the UTCM, which I personally feel a high cost with.
_____ 

> Chop Transfer Ejection.

This is brilliant! I like this HAHA. I totally forgot the chop transfer was a thing. (do people still use this?)
_____

> This [trash pull promise] Finesse can be Layered, even while in Bluff Seat.

This is super fun. Especially since cluing it directly could potentially be a save. Does this get confusing with contextual reads on 1s and the unnecessary trash bluff chop move? Also curious how likely a double finesse is compared to a UTCM. The situations that would allow for a tppf are identical to those for a UTCM, except now the cards have to match for it to be legal. Funky!
_______

> Unnecessary Trash Composition

Amusingly, I feel like my brain is too small for this. Trash bluffs/finesses + CMs get unrelated cards to play, and I think are actually valuable enough to be chosen over cluing the playable on chop directly?

I feel like this is an example of where it's not actually weird to choose the trash option over the direct clue to the playable. 

The composition for chop moving trash is fun though. Get Bob to finesse and Cathy to eject.
_______

> Dream Catcher Charm

And if a charm position no longer exists, is this an un-defined clue? Or are we rewriting unnecessariness here. 
Also, this name is fun! Would love to hear some of the inspiration for the nomenclature. 
______

Lastly, I will note that this framework overlaps with an already-existing elegant one, which is:
- if known trash is used for something unnecessary -> blind-play
- if unknown trash is used for something unnecessary -> cm
______

All around, some lovely thoughts to read. Would be super cool to extend this conversation more.

**Comment by timotree3** [comment]
1. Thanks for the interesting write-up!!
2. I feel like it happens pretty often that in review an Unnecessary Trash Chop Move is found to be the best clue in a position
3. While in some ways this is more pattern-oriented and simple than before, the removal of UTCM does break the previous pattern that if someone uses a trash chop move/trash push when they could've clued the card directly then it always calls for a slot 1 play.
4. The section on Delayed Double Pulls was a little confusing. Here's how I might re-write it:

- We propose that most CM Pulled cards can be Delayed in the same way as a normal Pull. Reverse Finesse interpretations should still never be entertained.
- However, there are some cases where if the card was Delayed Playable, Bob would not have blind-played because it looks like the clue was necessary in order to save both cards.
    - For example:
        - Donald holds a known g1
        - Alice uses the team’s last clue to 5 CM Cathy’s g2
        - Bob sees that if Alice had clued the g2 directly, Cathy might've discarded the 5, therefore Bob does not blind-play.
- Therefore, whenever a CM Pull occurs, the recipient should ask themselves "If my card was Delayed Playable, would Bob have blind-played?"
- If the answer is no, for example, if the 5 CM recipient was already “loaded” with either a play or a safe discard, then it is safe to play the Pulled card immediately.

**Comment by Zamiell** [comment]
kyle says

> "If a single, playable card is ever Chop Moved, the next player should immediately blind-play a card. Afterwards, the clued player can now always consider this card as being Pulled."
I've been in many games where a card is chop moved because the duplicate of it is on the table and it will play for free when the team kills the other copy. This seems, anecdotally, like a valuable tool to keep. Additionally, in low clue situations 5 chop moves on playable are implemented knowing the the card will play for free later in the game.
In the early/lowscore phase of the game, good deals will often leave the team with an abundance of playable cards and not enough clues to get them immediately. 1-3 of a color might be clued, and the 4 of that color needs to be saved with a 5cm using the last clue. Requiring a slot 1 to now bomb in order to save the card seems like a situation likely to arise.
"all one-off-chop 5 rank clues are to be promised as 5 Chop Moves, regardless of the identity of the 5." 
It seems ridiculous that a play clue to a playable 5 would also have to be assumed to have a chop move component. One of the basic tenants of the hyphen-ated conventions is that a clue has one and only one focus and/or meaning. A play clue to a known playable 5 should first and foremost be a play clue.
"ambiguity of Unnecessary Trash CMs (UTCM) as written are quite frustrating and therefore are very rarely performed." 
I strongly disagree with the claim of ambiguity, and softly disagree with the frequency in which they are performed.
"Therefore this Finesse can be Layered, even while in Bluff Seat."
Seems odd to have one unique convention that now does not have a bluff seat associated with it.

**Comment by Zamiell** [comment]
not being able to chop move playable cards in low-clue-count-situations does seem like it would be a semi-common problem

**Comment by florian-5f3759df** [comment]
I think the proposal with turning a trash chop move on a playable into a kind of 'pull' is an improvement. Currently that interpretation is taken up by the unnecessary trash chop move on a critical that you could clue directly, but I would bet that those opportunities are less common, because more cards are playable than savable, and that this could be a pretty common way to get a free extra card. Although the current old interpretation did get me out of a tight spot a few times.

(In some sense, I feel that most consistent with other conventions would be to have a bad chop move discharge, an ejection if you trash CM a playable card - in analogy with the rank choice ejection - and slot 1 if you unnecessarily save a critical card with trash CM instead of directly. But that's probably worse in practice, since when you can do a bad cm discharge you often also have a UTD available, so less diverse slot access, so I'm in favor of this one.)

For the 5s, I like it too, but it would be nice if there's a simple line to draw, since doing that for legitimate reasons is way more common than the trash version. 


> Seems odd to have one unique convention that now does not have a bluff seat associated with it.

I don't think it's unique, since the proposal is to do it exactly like we play the (layered) 5 pull. There we also allow a layered finesse to occur, even in bluff seat, as long as the first blind play is forward.

**Comment by florian-5f3759df** [comment]
In the proposal you make the current suboptimal save using trash instead of clue a charm. It seems to me that charms are not so useful in general (although admittedly we don't have many, so maybe I'm underestimating them). Maybe we can consider the following alternative for that clue:

If you save a critical card with trash instead of cluing it directly, and you could save it without bad touch, it becomes a finesse / play clue towards that card (of which the first blind play has to be forward). Currently suboptimal saves can turn saves into play clues, but rank choice and suboptimal saves are very hard to interpret -- very commonly neither of the options is clearly suboptimal. Saving with a TCM is then obviously suboptimal, and having the clue receiver play is very compatible with this proposal already.
(No bluffs allowed, except as exotic double bluff in 4p/5p.)

Example (all 1s are played, r3 is in trash):
Bob: r2 x x x
Cathy: b1 r3 r(5) (b)3

Neither 3 nor red would be a suboptimal save, since both give Cathy useful information. So Alice clues 1 to Cathy. Now Bob plays slot 1, since this is clearly suboptimal. Afterwards, Cathy thinks she maybe had b2 or y2 on chop, but after playing it, it turned out to be r3.

**Comment by pianoblook** [comment]
quick apology for not responding to stuff this week. I've had my hands full and have wanted to give all these a proper amount of time and thought

**Comment by pianoblook** [comment]
@Jayhui-q I added a bunch of in-game examples. I'll repost it down here too:

5CM Double:
https://hanab.live/replay/403684#17
https://hanab.live/replay/329695#18
https://hanab.live/replay/334096#22
https://hanab.live/replay/295971#37

Trash Double Pull:
https://hanab.live/replay/380594#45
https://hanab.live/replay/352324#42
https://hanab.live/replay/310831#36
https://hanab.live/replay/290377#37
https://hanab.live/replay/274009#39 (half credit since all critical are seen)

Trash Delayed Double Pull:
https://hanab.live/replay/413161#40

Midgame Multi-TCM Double Pull:
https://hanab.live/replay/381468#24
https://hanab.live/replay/295934#37

TCCM Double Pull:
https://hanab.live/game/17625#33
https://hanab.live/replay/316999#32

(attempted) stale TCM Finesse
https://hanab.live/replay/396259#15

1-away 5 BCME
https://hanab.live/replay/310789#39

BTCCME:
https://hanab.live/replay/305868#34

RBOCME:
https://hanab.live/replay/289454#33

**Comment by pianoblook** [comment]
from tonight: _Composition_ triple play: https://hanab.live/replay/427404#26

**Comment by pianoblook** [comment]
Replying to @Jayhui-q 

> Overall, I think there's definitely a lean towards faster play and away from flexible play.

It's a lean towards more efficient play. I'd argue it's a step towards more flexible play as well, as long as you don't rush. Current conventions hold a significant (and dare-I-say unhealthy) reliance on needing to get cards blind-played immediately while on slot 1. One of the main strategic benefits of this system is that it creates more opportunities to _Eject_ or _Pull_ efficiently + effectively throughout the game. 

Overall, my argument is that this leads to a lot *more* options, on a macrolevel. I already tried to cover the 3 individual scenarios where trouble could lie in the doc. Overall, I firmly believe that the *frequency* of when those situations would actually be an impediment are vastly outweighed by the *frequency* of situations in which this system would be far superior.

Now, what **would** be rushed and inflexible is to then take this increased freedom as a way to discard more aggressively. While the possibility for this behavior is certainly there, that's just a losing strategy outside of tournament play / high-efficiency variants. But no, my goal is to give *more* opportunities to slow down and, say, TCM a 1- or 2-away card; or feel free to start playing out your hands without concern over immediately 'getting' a new card; or having more KT with which to _Shout_ at non-criticals; etc etc. 

> I think 5cm's are actually pretty well defined[...]

Please reread the doc section on this: I try to lay out a bunch of possible edge-cases that can complicate off-chop 5 clues, *especially* contextually vs text-as-written. Overall the system as-is obviously functions... my argument is just that it's suboptimal. The main problem comes from the *theoretical* existence of a 1-away 5 once you're into midgame scenarios. It means (a) you can never BCME with a 1-away 5; and (b) 5 double-pulls **always** become illegal after because the focus of the clue would always just be the 5 after any blind-play.

In other words, this new system basically just asks for one small change: "ignore how close to playable the 5 is". I know that feels weird, but I do try my best to explain how remarkably unlikely it is for that change to cause any harm. the *benefits*, however, are massive.

> Don't we all [UTCM a critical] all the time? We just deal with it and clarify which one was the good one later. I may or may not be alone in this practice.

I don't think I've *ever* done this move. Maybe once in an emergency situation. As Zamiel always says, a cm is a 0-for-1. So sure, UTCMs as a save are fine...as a 1-for-1. But see my earlier point about the goal of freeing us from needing to give these sorts of 1-for-1 in the first place. Just save the card and move on! That said, of course I have fully conceded that there are legit situations in which a UTCM would be optimal. Every convention has a tradeoff.

> In my experience, Bad TCCM ejections have caused more strikes than they were worth. 

Sure, but only because they're currently undefined, confusing, and untrustworthy. But to very clear: these only work when the tempo (stall) clue would be illegal. And I've personally never seen someone feel the urge to (illegally) tempo clue a single card. Also fwiw, BTCCMEs feel *great* - getting an Ejection + Tempo + KT is a wonderful combo.

> First, I think it has to be "directly playable" rather than a delayed play clue, as pulls are[...] I don't think we would rather give the direct play and then give the 5, because it's less efficient[...]

I'm totally fine restricting 5 double pulls to being for only directly-playables. As others have pointed out - and I discussed in the doc, too - there absolutely would have to be restrictions in place if we wanted to 'pull off' delayed 5CM pulls. 

Basically, I'd highly suggest that midgame 5DPs should just function exactly the same as early game 5DPs. Are delayed 5DPs allowed in early game? Probably only if it is clear and safe, also.

And in general, I'd ask you and others this: are 5 Double Pulls good in early-game? If so, why are they suddenly **bad** in mid-game? IMO if y'all are against midgame 5DPs because of these strange edge-cases, perhaps you might want to submit a PR to remove them altogether :P

> b2 is on the stacks
r4, g3 is in the trash
Bob has an unknown 4 (it is b4)
Cathy's hand is b4 - x - r5 - b3
I think Alice giving a 5cm is far superior to blues as a direct play clue. In this framework, 5s would cause Bob to blind-play.

This sort of edge-case happens in early-game 5 double pulls too, right? e.g. replace your example with the following:

b1 is on the stacks
Cathy's hand is b1 - x - r5 - b2

No one should RCE or 5DP from a 2 clue or 5 clue here, since the direct play clue *was not possible anyway*. This just seems like normal context to me, no?

I also disagree that 3 would be the optimal clue here. I'd want to clue 3 and have Bob follow up with 5. Therefore: (a) the 3 immediately plays (declined 5cm) and (b) the b4 is known (declined blue clue). In *general*, remember, a main reason why it can be disadvantageous to CM something is that it will no longer be focused. So actually, just remove Bob's knowledge about a 4 in his hand and you can still do a double pull here: cluing 5s is atrocious compared to just blue, yes?

> In addition, more stipulations for ambiguity must be made. If Donald has m1 that is finessable, the blue clue is also bad.

Seems like a plus for the double pull camp, lol. I mean blue still works, you just need to also 'get' Donald's m1. But yes, much better tempo if you do the pull immediately!

**Comment by Jayhui-q** [comment]
Thanks so much for the detailed responses! Seriously, massive kudos 🔥.

> Basically, I'd highly suggest that midgame 5DPs should just function exactly the same as early game 5DPs. Are delayed 5DPs allowed in early game? Probably only if it is clear and safe, also.

Agree with this. 

Just in general, I agree with a lot of points you brought up. My hope is that I voice these concerns so as to refine the idea discourse and I would be very happy with the proposal going either way. Massively appreciate all the work you put into both the idea generation and the discussion.

**Comment by pianoblook** [comment]
Replying to @timotree3 

> Contrary to your experience, I feel like it happens pretty often that in review an Unnecessary Trash Chop Move is found to be the best clue in a position

The question at hand is just this: **which set of conventions would lead to better outcomes across many games?**.  A large thrust of what the doc is trying to argue is just that there are a lot of benefits to restricting this move to promising a playable. This is in terms of simplicity, efficiency, tempo, clarity, *and* opportunity to expand into other conventions (notice how most of the 'extensions' included relate to trash moves!)

> While in some ways this is more pattern-oriented and simple than before, the removal of UTCM does break the previous pattern [of Unnecessary moves always calling for] a slot 1 play.

Agreed! It does break that pattern - to which I say thank goodness 😆 I didn't want to lean too far into this point in this proposal, but yes I do believe (as @florian-5f3759df has brought up in the past too) that there's room to improve the _Unnecessary Trash_ algorithm to better relate to the situation at hand. 

For example, see the _UT Composition_ extension which allows for Cathy to Eject, Finesse, or not react in response to a UT clue. For an example of this already being utilized, check out the example [above](https://github.com/hanabi/hanabi.github.io/issues/479#issuecomment-774401660).

> 4. The section on Delayed Double Pulls was a little confusing. Here's how I might re-write it: [...]

Cool, I like that! Yeah the existence of Delayed Double Pulls seems to be a fairly contentious bit at the moment, but I liked the way you phrased all that. 

FWIW, if we do allow delayed Pulls, I believe(?) trash CMs could basically **always** be safely respected as potentially-delayed. The problem comes from 5CMDPs, since obviously then Bob and Cathy would need to consider whether it might have been dangerous to clue the card directly.

**Comment by pianoblook** [comment]
Responding to @Zamiell / Kyle:

 
> I've been in many games where a card is chop moved because the duplicate of it is on the table and it will play for free when the team kills the other copy. This seems, anecdotally, like a valuable tool to keep. 

Is it currently legal to 5 double pull a card when the copy is seen elsewhere? I don't know of any such restriction, and if it is allowed then I suppose why should it be any different outside of early game? But maybe it should be - that's fine too; we can just create that exception for early- and mid-game 5DPs alike.

> Additionally, in low clue situations 5 chop moves on playable are implemented knowing the the card will play for free later in the game.

I covered this in the doc: yes, in some contexts a 5CM should be respected as 'necessary' if clues are low.

> In the early/lowscore phase of the game, good deals will often leave the team with an abundance of playable cards and not enough clues to get them immediately. 1-3 of a color might be clued, and the 4 of that color needs to be saved with a 5cm using the last clue. Requiring a slot 1 to now bomb in order to save the card seems like a situation likely to arise.

Covered this exact situation in the doc. I agree 100%. What do you think of timo's suggested [rewrite](https://github.com/hanabi/hanabi.github.io/issues/479#issuecomment-769544794) of that section?

> It seems ridiculous that a play clue to a playable 5 would also have to be assumed to have a chop move component[...] A play clue to a known playable 5 should first and foremost be a play clue. 

Hopefully it seems less ridiculous after reading through my section discussing that. Just Push it, clue it once it's on Chop, or even trash double-pull it as soon as someone else draws a playable ;)

> One of the basic tenants of the hyphen-ated conventions is that a clue has one and only one focus and/or meaning. 

Agreed! All I'm pushing for is to change the focus to **always** being the cm'd card.

> I strongly disagree with the claim of ambiguity [of UTCMs], and softly disagree with the frequency in which they are performed.

I'm not sure what claim there is to disagree on, in terms of ambiguity. There's no disputing that the TCM'd card in such an instance can always either be playable, delayed playable, or critical-and-directly-savable. So if we adopted a system in which Bob only blind-played in cases where it's playable, that would necessarily be less ambiguous.

> Seems odd to have one unique convention that now does not have a bluff seat associated with it.

Actually this is already exactly the same way 5-Pull finesses work: if the first blind-play doesn't match the _Pulled_ card, you're **promised** that card deeper in your hand (bluff seat or not)

**Comment by pianoblook** [comment]
responding to @florian-5f3759df 

> I think the proposal with turning a trash chop move on a playable into a kind of 'pull' is an improvement. 
🥳 🎈 

> For the 5s, I like it too, but it would be nice if there's a simple line to draw, since doing that for legitimate reasons is way more common than the trash version.

Sounds like everyone seems to be reaching that consensus around 5s: that sometimes a 5 double pull shouldn't be assumed if there are good reasons to 5cm it anyway.

Perhaps two things could happen then:
1) 5 Double Pulls (as written) should be updated to reflect this potential risk. Early game or not, it sounds like there are a few legitimate times when a "Pull" should not be initiated
2) Treat mid-game 5DPs with the same exact logic

------
2nd comment:

> Maybe we can consider the following alternative for [the proposed Charm][...]

Extension 2: **Trash Pull Promise Finesse** actually describes exactly the situation/example you wrote :D

The _Charm_ version would only be if Bob sees that the _CM_'d card is **2+ away from playable**

**Comment by pianoblook** [comment]
Forgot to post this one - reverse (zero clues) double tccm pull

https://hanab.live/replay/426134#34

**Comment by Zamiell** [comment]
does this conflict with #460

**Comment by pianoblook** [comment]
> does this conflict with #460

I don't believe so! That's more focused on recluing trash/hard burns, while this focuses on clues that actively + intentionally _Chop Move_ stuff

**Comment by pianoblook** [comment]
https://hanab.live/replay/455134#6

OOP + delayed 5 double pull 😆 #cluestarvedthings

**Comment by Zamiell** [comment]
it looks like there is a lot of evidence posted for this move
it is easier to evaluate what is changing if you submit a minimalistic PR.
can you submit a pr with just the base stuff and not any of the new moves (e.g. no dream catcher finesse, that belongs in a new issue)
then we can iterate on that to see if there is anything problematic

**Comment by pianoblook** [comment]
Nice 5BCME even though the 5 is globally known playable https://hanab.live/replay/468067#20

**Comment by pianoblook** [comment]
> can you submit a pr with just the base stuff and not any of the new moves (e.g. no dream catcher finesse, that belongs in a new issue)
> then we can iterate on that to see if there is anything problematic

jeez sorry I missed this notification, too - not sure why >.<

Anyway, my intention for this github was to *only* propose the 3 main principles. So to clarify, I should go ahead and make a PR for the main 3 changes?

**1. Unnecessary Chop Move Principle**

**2. Bad Chop Move Ejection Principle**

**3. 5 Chop Move Principle**


Also, what level would you suggest putting each of these?

**Comment by Zamiell** [comment]
i would not formulate these are principles, i would formulate them as moves

e.g.

1) change BCME section to say that it can be triggered by any type of chop move that chop moves trash
2) add new section called "Musical Ejection" that says its triggered by any type of chop move that chop moves a playable card
3) no changes need to be made to the doc ???

**Comment by Zamiell** [comment]
oh and probably #4 - delete the 5 double pull from the doc

**Comment by Zamiell** [comment]
> adopting these would technically lead to a decrease in overall conventions: Trash Pull

how does this interact with a trash pull

**Comment by sjdrodge** [comment]
I'd personally like an exception for Unnecessary Chop Move Principle for 5CMing playables when it's plausible that the chop-moved player might discard on a direct play clue. Also, if not already made clear, it should only be considered unnecessary to CM a playable if you can cleanly clue it.

In other words, I think we should include some criteria for when chop moving a playable card is considered Unnecessary.

**Comment by Zamiell** [comment]
> I'd personally like an exception for Unnecessary Chop Move Principle for 5CMing playables when it's plausible that the chop-moved player might discard on a direct play clue

agreed, bake that it

> it should only be considered unnecessary to CM a playable if you can cleanly clue it.

agreed, bake that in

**Comment by pianoblook** [comment]
@sjdrodge I believe most of those concerns were covered via #491 - definitely agree that nothing fancy should trigger if the clue might just be 'necessary' to keep everything safe!

Zam I want to be sure we're on the same page about all this stuff. Let's see:

> 1. change BCME section to say that it can be triggered by any type of chop move that chop moves trash

- Does that mean you're comfortable allowing stuff like Bad Tempo Clue CMs + Bad Order Chop Moves?
- I hope so, since indeed that would make for an incredibly simple + universal convention that covers everything!

> 2. add new section called "Musical Ejection" that says its triggered by any type of chop move that chop moves a playable card

What's being proposed is that Bob should blind-play, not Eject; in fact Ejecting would promise the clued player that the CM'd card is trash, not playable. 

I like the idea of making a special name for it; then we could indeed have it all fall under one umbrella 'move'.

> 3. no changes need to be made to the doc for 5 Chop Move Principle ???

What's being proposed does pretty drastically expand/alter 5 Chop Moves though. Games played with Hyphen-ated conventions as-is can have plenty of situations in which a one-off-chop 5 clue triggers a direct Bluff or Finesse on the 5 itself. 5 CM Principle would change that to **always** be chop move, until late game ofc

The thing is, midgame 5 Double Pulls (or 'musical chop move' or w/e we end up calling it) can never exist if a rank 5 clue could also be interpreted as a direct Bluff on the 5.

> how does this interact with a trash pull

I guess not all trash pulls are actually unnecessary, so you're right that Trash Pull should remain as its own convention

**Comment by Zamiell** [comment]
> Does that mean you're comfortable allowing stuff like Bad Tempo Clue CMs

a tccm on trash would cause a bcme

> Does that mean you're comfortable allowing stuff like Bad Order Chop Moves?

a ocm on trash would cause a bcme.
but you can only do it forward and you can't do it in reverse

> What's being proposed is that Bob should blind-play, not Eject; in fact Ejecting would promise the clued player that the CM'd card is trash, not playable.

sorry, you're right, so add a new section called Musical Ignition

> why Musical, I wonder?

because https://www.thesaurus.com/browse/play?r=76&src=ref&ch=the

> Games played with Hyphen-ated conventions as-is can have plenty of situations in which a one-off-chop 5 clue triggers a direct Bluff or Finesse on the 5 itself. 5 CM Principle would change that to always be a chop move first-and-foremost, regardless of the exact identity of the 5. 

oh
basically you are saying that 5bcme takes precedence over a one-away finesse.
i doubt most people are on board with that.
unless you are in the specific situation where there is free choice between a color clue and a number clue - then its fine.

> midgame 5 Double Pulls (or 'musical chop move' or w/e we end up calling it) can never exist if a rank 5 clue could also be interpreted as a direct Bluff on the 5.

i think 2-for-1s are good enough such that we don't have to always configure it as a 3-for-1.
rather, it is only a 3-for-1 in the situation where free choice exists.
remind me if there was any discussion above about this specifically

**Comment by sjdrodge** [comment]
How about we just say that if you can give a trash push on the 5 and you clue 5 instead, it's BCME. If there's no trash push available for the 5, one-away finesse takes precedence.

**Comment by Zamiell** [comment]
to be more clear, the criteria should be
1) a direct color clue
2) a trash push

**Comment by pianoblook** [comment]
Unfortunately I don't think it really works to hybridize. Stephen's idea would work fine for determining between a BCME vs direct finesse/bluff, but it doesn't work when the chop card is playable.

Let's say Cathy gets a 5 clue, and Bob blind-plays slot 1. The issue is that Cathy won't be able to tell if the blind-play is from a  _Musical Ignition_ (still might try and workshop that name), or if it was a _Bluff_ (or, hell, a true _Finesse_!) on the 5 itself.

I described these issues in detail in the linked doc, so I'd urge everyone to read the relevant section. In fact, I'm gunna post it here:

> **Change 1: Rank 5 Finesses/Bluffs are now impossible when one off-chop**
This might seem like a fairly significant loss at first glance. While it’s certainly true that this may lead to less efficient lines in certain cases, fortunately it turns out to be much less common, (and damaging) than it appears:
A. Most of the time, the same Bluff or Finesse can still be triggered by cluing color to the 5.
B. When a color clue is blocked, and the chop *is* trash, we can sometimes still work around this by performing a Trash Push Finesse (but not always).
C. What if the Chop card isn’t trash? If it’s directly playable, then this new 5 Double Pull interpretation is even stronger than the normal interpretation. And of course if it’s just some other useful-but-non-playable, no blind-play would be called for anyway.

> This does leave a couple legitimate scenarios in which you’d feel the loss of not being able to use a rank clue for finesse: (a) wanting to Bluff using a 1-away 5 while color is blocked and the chop is exactly trash, or (b) wanting to perform a Double Finesse with a Reverse Component - but once again only if color is blocked and the chop is exactly trash. 
> These last remaining hypotheticals aren’t very compelling cases. For example, simply Bluff some other way, or hold your nose and settle for the 2-for-1 Finesse, and just clue the 5 later. Also, notice how all these situations necessarily involve the 5 not being in danger, since a trash card must be on chop to begin with!

**Comment by pianoblook** [comment]
The simple way of stating this change is just, "until Late Game, a 5s rank clue can *never* be given as a direct Play Clue. Instead - even in midgame - it will always either be a _5s Chop Move_ or _5 (Double) Pull_

**Comment by sjdrodge** [comment]
> The simple way of stating this change is just, "until Late Game, a 5s rank clue can _never_ be given as a direct Play Clue. Instead - even in midgame - it will always either be interpreted as a _5s Chop Move_ or _5 (Double) Pull_"

This is for 5's that are 1-away from chop, right? 2-away is still governed by LSP?

**Comment by pianoblook** [comment]
> > The simple way of stating this change is just, "until Late Game, a 5s rank clue can _never_ be given as a direct Play Clue. Instead - even in midgame - it will always either be interpreted as a _5s Chop Move_ or _5 (Double) Pull_"
> 
> This is for 5's that are 1-away from chop, right? 2-away is still governed by LSP?

Oh, yes good point! Right, let me rephrase:

"until Late Game, a 5s rank clue can _never_ be given as a direct Play Clue on a 5 that is one-away from chop. Instead - even in midgame - it will **always** be interpreted as a _5s Chop Move_."

**Comment by pianoblook** [comment]
in fact, this is actually exactly what is written in Principle 3 lol:

> **3. 5 Chop Move Principle**
Until the End-Game, except for valid stalling situations, all one-off-chop 5 rank clues are to be promised as 5 Chop Moves, regardless of the identity of the 5.

**Comment by Zamiell** [comment]
> Let's say Cathy gets a 5 clue, and Bob blind-plays slot 1. The issue is that Cathy won't be able to tell if the blind-play is from a Musical Ignition (still might try and workshop that name), or if it was a Bluff (or, hell, a true Finesse!) on the 5 itself.

oh, i see

> The simple way of stating this change is just, "until Late Game, a 5s rank clue can never be given as a direct Play Clue. Instead - even in midgame - it will always either be interpreted as a 5s Chop Move or 5 (Double) Pull"

ok then that should be in the 5's Chop Move section.

**Comment by sjdrodge** [comment]
I have to admit I'm pretty skeptical of not allowing finesses w/ a number 5 clue in late midgames when the 5 clue is otherwise blocked. I'm not sure how often we're getting these 3-for-1 Double Pulls after LSP anyway, and that is the main gain. (The 2-for-1 BCME we'd still have access to whenever a Trash Push was available or the color clue was unblocked.)

**Comment by pianoblook** [comment]
There's certainly room for debate around which is more useful. Personally I think there's a good amount of value in the simplicity that comes from just always trusting it'll CM ('badly' or not). And at least in my experience I can't remember the last time I've ever felt the urge to rank-clue a 5 in that fashion 🤷 Feels like a very rare hypothetical that we're discussing, tbh.

The tradeoff, I guess, is either [allowing more BCMEs + Double Pulls] vs [allowing more rank 5 bluffs/finesses].

**Comment by sjdrodge** [comment]
Let's make sure we're discussing the pros/cons with full clarity.

First let's consider the case where the color clue to a 1-away 5 is blocked, but the card on chop is cleanly cluable as KT:
- With the status quo, it is ambiguous for Bob whether a 5 clue is a potential bluff, or a BCME
- With the proposed change, it is always a BCME.
- With the proposed compromise position, it is always BCME.

Next let's consider the case where chop is a playable card (and we've left LSP):
- With the status quo, it is potentially ambiguous for Cathy whether a finesse/bluff has happened or chop is playable
- With the proposed change, it is always a 3-for-1 Double Pull
- With the proposed compromise position, we have the same ambiguity as the status quo

Last let's consider the case where both the color clue is blocked, and the chop card is trash but not cleanly cluable as KT:
- With the status quo, it is ambiguous for Bob whether a 5 clue is a potential bluff, or a BCME
- With the proposed change, it is always a BCME
- With the proposed compromise position, it is always a finesse/bluff

The compromise position has a significant edge in case 3, imo. Case 2 is much simpler using the original proposal, but I'm not certain if that actually gives it the edge. Often times Alice and Cathy will be able to see enough 5's (or there's simply enough information on Cathy's 5) such that there's no ambiguity. There's definitely a lot to be said for the simplicity though.

**Comment by sjdrodge** [comment]
As for rarity. I agree it's not the most common thing in the universe, but my experience is that it comes up fairly often that play clues are given w/ number 5 in the late mid-game. I strongly suspect that the 3-for-1 opportunities in non-LSP mid-game are rarer.

**Comment by sjdrodge** [comment]
Screw it though, I'll vote for the proposal as-is. Simplicity is king.

**Comment by pianoblook** [comment]
Good idea to lay out all the situations side-by-side, that's helpful.

I think I agree with you that in most games it's going to be less rare to find situation C (chop is trash but push & color are blocked) than situation B (playable on chop).

That said, I think the fact that Cathy's chop is always trash in those situations makes it seem less *impactful* to build the conventions with that situation in mind. So in other words, even if Situation C may be somewhat more common, I'd rather have an efficient fail-safe way of dealing with Situation B when it does arise, if that makes sense?

**Comment by sjdrodge** [comment]
This issue has been open for quite some time, and it looks to me like the consensus is thumbs up. Time to /accept?

**Comment by Zamiell** [comment]
read 612

**Comment by sjdrodge** [comment]
Ah I see.

**Comment by Zamiell** [comment]
summary of this thread:

- delete 5 double pull
- add playful ignition
- add playful ignition interaction

piano, is that right?

**Comment by pianoblook** [comment]
> summary of this thread:

Basically as follows:

- add playful ignition + 5 interaction [done]
- expand BCMEs [done]
- reimagine _Unnecessary Trash CM_ [currently in a gray area; my plan was to create a new git issue after this first batch was finalized]
